tests/testsuite/standard_lib.rs

```diff
@@ -77,10 +80,48 @@ fn setup() -> bool {
         .add_dep(Dependency::new("rustc-std-workspace-std", "*").optional(true))
         .feature("mockbuild", &["rustc-std-workspace-std"])
         .publish();
-    return true;
+
+    let p = ProjectBuilder::new(paths::root().join("rustc-wrapper"))
+        .file(
+            "src/main.rs",
+            r#"
+                use std::process::Command;
+                use std::env;
+                fn main() {
+                    let mut args = env::args().skip(1).collect::<Vec<_>>();
+
+                    let crates = [
+                        "alloc",
+                        "compiler_builtins",
+                        "core",
+                        "panic_unwind",
+                        "proc_macro",
+                        "std",
+                        "test",
+                    ];
+                    let is_sysroot_crate = args.iter()
+                        .any(|a| a.starts_with("rustc_std_workspace") || crates.iter().any(|b| a == b));
+
+                    if is_sysroot_crate {
```

Comment:
This can actually be simplified now I think to testing for `RUSTC_BOOTSTRAP` perhaps? (maybe with some extra checks for `--target` being passed as well)